### PR TITLE
Npm implementation

### PR DESCRIPTION
**Pull request by andreespirela**
## What kind of change does this PR introduce?

This PR introduces NPM modules.

```
import {} from "npm:package"
```

Dependent on #188

### CODE REVIEW

**Comment by haschu** [comment]
Awesome work! Can't wait for this! 😍

File: crates/base/src/deno_runtime.rs [hunk 1]
```diff
@@ -429,15 +444,35 @@ mod test {
         let specifier = binding.to_str().unwrap();
         let format_specifier = format!("file:///{}", specifier);
         let module_specifier = ModuleSpecifier::parse(&format_specifier).unwrap();
-        let create_module_graph_task = create_graph_and_maybe_check(vec![module_specifier]);
+        let builder = ModuleGraphBuilder::new(
+            Some(Arc::new(Mutex::new(
+                Lockfile::new(
+                    PathBuf::from(
+                        "/Users/andrespirela/Documents/workspace/supabase/edge-runtime/deno.lock",
```
Comment (laktek) [inline crates/base/src/deno_runtime.rs:451]: hardcoded path

**Review by laktek — APPROVED** [review]
Ship it! 🚢

**Comment by laktek** [comment]
@andreespirela Do you want to sqaush some of the commits? Make sure to have a `feat:` commit, so it will create a new release.

**Comment by laktek** [comment]
@andreespirela Do you want to sqaush some of the commits? Make sure to have a `feat:` commit, so it will create a new release.

File: crates/npm/tarball.rs [hunk 2]
```diff
@@ -131,7 +131,7 @@ fn extract_tarball(data: &[u8], output_folder: &Path) -> Result<(), AnyError> {
                 // symlinks to the npm registry. If ever adding symlink or hardlink
                 // support, we will need to validate that the hardlink and symlink
                 // target are within the package directory.
-                log::warn!(
+                println!(
```
Comment (laktek) [inline crates/npm/tarball.rs:134]: should this be a log::warn?

File: crates/npm/resolution.rs [hunk 3]
```diff
@@ -293,7 +292,7 @@ async fn add_package_reqs_to_snapshot(
             .iter()
             .all(|req| snapshot.package_reqs().contains_key(req))
     {
-        log::debug!("Snapshot already up to date. Skipping pending resolution.");
+        println!("Snapshot already up to date. Skipping pending resolution.");
```
Comment (laktek) [inline crates/npm/resolution.rs:295]: should all `println` use `log::debug` or `log::warn`?

**Pull request merged**